src/VecSim/algorithms/hnsw/hnswlib.h

```diff
@@ -891,42 +888,51 @@ void HierarchicalNSW<dist_t>::SwapLastIdWithDeletedId(tableint element_internal_
         this->entrypoint_node_ = element_internal_id;
     }
 }
-
+/* typedef struct {
+    VecSimType type;     // Datatype to index.
+    size_t dim;          // Vector's dimension.
+    VecSimMetric metric; // Distance metric to use in the index.
+    size_t initialCapacity;
+    size_t blockSize;
+    size_t M;
+    size_t efConstruction;
+    size_t efRuntime;
+    double epsilon;
+} HNSWParams; */
 template <typename dist_t>
-HierarchicalNSW<dist_t>::HierarchicalNSW(SpaceInterface<dist_t> *s, size_t max_elements,
-                                         std::shared_ptr<VecSimAllocator> allocator, size_t M,
-                                         size_t ef_construction, size_t ef, size_t random_seed,
-                                         size_t pool_initial_size)
-    : VecsimBaseObject(allocator), element_levels_(max_elements, allocator),
-      label_lookup_(max_elements, allocator)
+HierarchicalNSW<dist_t>::HierarchicalNSW(const HNSWParams *params,
+                                         std::shared_ptr<VecSimAllocator> allocator, size_t ef,
```

Comment:
Can't you get `ef` from the params?

Reply:
ef runtime wasnt passed by the original bf wrapper ctor and was always set to the default value defined by hanswlib ctor params, so I didn't change it

Reply:
you are, i can